test/jdk/tools/jar/JarExtractTest.java

```diff
@@ -0,0 +1,382 @@
+        final int exitCode = JAR_TOOL.run(new PrintStream(outStream), System.err, "--help");
+        Assert.assertEquals(exitCode, 0, "jar --help command failed");
+        final String output = outStream.toString();
+        final String expectedMsg = "--dir                    Directory into which the jar will be extracted";
+        Assert.assertTrue(output.contains(expectedMsg), "jar --help didn't contain --dir option");
+    }
+
+    /**
+     * Tests that {@code jar -x -f} command works fine even when the -C or --dir option
+     * isn't specified
+     */
+    @Test
+    public void testExtractWithoutOutputDir() throws Exception {
+        final int exitCode = JAR_TOOL.run(System.out, System.err, "-x", "-f", testJarPath.toString());
+        Assert.assertEquals(exitCode, 0, "Failed to extract " + testJarPath);
+        // the content would have been extracted to current dir
+        verifyExtractedContent(".");
+    }
+
+    /**
+     * Tests that extracting a jar using {@code -P} flag and without any explicit destination
+     * directory works correctly if the jar contains entries with leading slashes and/or {@code ..}
+     * parts preserved.
+     */
+    @Test
+    public void testExtractNoDestDirWithPFlag() throws Exception {
+        // create a jar which has leading slash (/) and dot-dot (..) preserved in entry names
+        final Path jarPath = createJarWithPFlagSemantics();
+        // extract with -P flag without any explicit destination directory (expect the extraction to work fine)
+        final String[] args = new String[]{"-xvfP", jarPath.toString()};
+        printJarCommand(args);
+        final int exitCode = JAR_TOOL.run(System.out, System.err, args);
+        Assert.assertEquals(exitCode, 0, "Failed to extract " + jarPath);
+        final String dest = ".";
+        Assert.assertTrue(Files.isDirectory(Paths.get(dest)), dest + " is not a directory");
+        final Path d1 = Paths.get(dest, "d1");
+        Assert.assertTrue(Files.isDirectory(d1), d1 + " directory is missing or not a directory");
+        final Path d2 = Paths.get(dest, "d1", "d2");
+        Assert.assertTrue(Files.isDirectory(d2), d2 + " directory is missing or not a directory");
+        final Path f1 = Paths.get(LEADING_SLASH_PRESERVED_ENTRY);
+        Assert.assertTrue(Files.isRegularFile(f1), f1 + " is missing or not a file");
+        Assert.assertEquals(Files.readAllBytes(f1), FILE_CONTENT, "Unexpected content in file " + f1);
+        final Path f2 = Paths.get("d1/d2/../f2.txt");
+        Assert.assertTrue(Files.isRegularFile(f2), f2 + " is missing or not a file");
+        Assert.assertEquals(Files.readAllBytes(f2), FILE_CONTENT, "Unexpected content in file " + f2);
+    }
+
+    /**
+     * Tests that the {@code -P} option cannot be used during jar extraction when the {@code -C} and/or
+     * {@code --dir} option is used
+     */
+    @Test
+    public void testExtractWithDirPFlagNotAllowed() throws Exception {
```

Comment:
I would include --extract in your testing options

Reply:
Done in latest version of the PR